File: Server/src/main/content/region/desert/quest/princealirescue/npc/LadyKeliNPC.java
package content.region.desert.quest.princealirescue.npc;

import core.game.node.entity.npc.AbstractNPC;
import core.game.node.entity.player.Player;
import core.game.world.GameWorld;
import core.game.world.map.Location;

/**
 * The Lady keli npc.
 */
public final class LadyKeliNPC extends AbstractNPC {

    private static final int[] ID = {919};

    /**
     * Instantiates a new Lady keli npc.
     */
    public LadyKeliNPC() {
        super(0, null, true);
    }

    private LadyKeliNPC(int id, Location location) {
        super(id, location, true);
    }

    @Override
    public AbstractNPC construct(int id, Location location, Object... objects) {
        return new LadyKeliNPC(id, location);
    }

    @Override
    public boolean isHidden(final Player player) {
        return player.getAttribute("keli-gone", 0) > GameWorld.getTicks();
    }

    @Override
    public int[] getIds() {
        return ID;
    }

    @Override
    public int getWalkRadius() {
        return 2;
    }

}

File: Emulator/src/main/content/region/desert/quest/rescue/handlers/LadyKeliNPC.kt
package content.region.desert.quest.rescue.handlers

import core.game.node.entity.npc.AbstractNPC
import core.game.node.entity.player.Player
import core.game.world.GameWorld.ticks
import core.game.world.map.Location

/**
 * Lady keli NPC.
 */
class LadyKeliNPC : AbstractNPC {

    constructor() : super(0, null, true)

    private constructor(id: Int, location: Location) : super(id, location, true)

    override fun construct(id: Int, location: Location, vararg objects: Any): AbstractNPC {
        return LadyKeliNPC(id, location)
    }

    override fun isHidden(player: Player): Boolean {
        return player.getAttribute("keli-gone", 0) > ticks
    }

    override fun getIds(): IntArray {
        return ID
    }

    override fun getWalkRadius(): Int {
        return 2
    }

    companion object {
        private val ID = intArrayOf(919)
    }
}
